frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDashboard.tsx

```diff
@@ -8,26 +8,30 @@ const REQUIRED_MEMBER_TEC_CREDITS = 3;
 const REQUIRED_LEAD_TEC_CREDITS = 6;
 
 const TeamEventCreditDashboard = (props: {
-  approvedTEC: TeamEventInfo[];
-  pendingTEC: TeamEventInfo[];
+  allTEC: TeamEventInfo[];
+  approvedAttendance: TeamEventAttendance[];
+  pendingAttendance: TeamEventAttendance[];
 }): JSX.Element => {
-  const { approvedTEC, pendingTEC } = props;
+  const { allTEC, approvedAttendance, pendingAttendance } = props;
 
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const userRole = useSelf()!.role;
 
   const requiredCredits =
     userRole === 'lead' ? REQUIRED_LEAD_TEC_CREDITS : REQUIRED_MEMBER_TEC_CREDITS; // number of required tec credits in a semester based on user role
 
-  const approvedCredits = approvedTEC.reduce(
-    (approved, teamEvent) => approved + Number(teamEvent.numCredits),
-    0
-  );
-  const approvedCommunityCredits = approvedTEC.reduce(
-    (communityCredits, teamEvent) =>
-      teamEvent.isCommunity ? communityCredits + Number(teamEvent.numCredits) : communityCredits,
-    0
-  );
+  let approvedCredits = 0;
+  let approvedCommunityCredits = 0;
+  approvedAttendance.forEach(async (attendance) => {
+    const event = allTEC.find((tec) => tec.uuid === attendance.eventUuid);
+    if (event !== undefined) {
+      const currCredits = attendance.hoursAttended
+        ? Number(event.numCredits) * attendance.hoursAttended
```

Comment:
Hm, the calculation for total credits is slightly different than down below. You check if `teamEvent.hasHours` in the implementation below for the TEC cards. In the implementation here, you check if `attendance.hoursAttended` evaluates to a truthy value. I like the below one better, since you also use the `getHoursAttended` helper that handles the attendance.hoursAttended being `undefined` case.